src/backend.rs

```diff
@@ -378,19 +381,49 @@ pub fn pass_outgoing_args(ctx: &mut Context, arity: u32) {
                     ; pop Rq(gpr)
                 );
             }
-            _ => unimplemented!("don't know how to pass argument {} via {:?}", arg_pos, loc),
+            ArgLocation::Stack(_) => {
+                let gpr = ctx.regs.take_scratch_gpr();
+                dynasm!(ctx.asm
+                    ; pop Rq(gpr)
+                );
+                stack_args.push(gpr);
+            }
         }
     }
+
+    let num_stack_args = stack_args.len() as i32;
+    dynasm!(ctx.asm
+        ; sub rsp, num_stack_args
+    );
+    for (stack_slot, gpr) in stack_args.into_iter().rev().enumerate() {
+        let offset = (stack_slot * WORD_SIZE as usize) as i32;
+        dynasm!(ctx.asm
+            ; mov [rsp + offset], Rq(gpr)
+        );
+        ctx.regs.release_scratch_gpr(gpr);
+    }
+
+    num_stack_args
 }
 
-pub fn call_direct(ctx: &mut Context, index: u32, return_arity: u32) {
+fn post_call_cleanup(ctx: &mut Context, num_stack_args: i32) {
+    dynasm!(ctx.asm
+        ; add rsp, num_stack_args
+    );
```

Comment:
It'll be pretty common that num_stack_args is 0, so it'd be a nice optimization to skip this instruction in that case.

Reply:
Already done on current HEAD (also, I fixed the problem where this doesn't multiply `num_stack_args` by `WORD_SIZE`)